File: app/src/androidTest/java/com/bbtest/perform/base/CpuThread.java
package com.bbtest.perform.base;

import androidx.test.uiautomator.UiDevice;

import com.bbtest.utils.CommonUtil;
import com.bbtest.utils.FileUtil;

import java.io.File;

public class CpuThread extends Thread {

    private UiDevice device = null;
    private int pid = 0;
    private File resultFile = null;

    private boolean isEnd = false;

    public CpuThread(UiDevice device, int pid, File resultFile) {
        this.device = device;
        this.pid = pid;
        this.resultFile = resultFile;
    }

    @Override
    public void run() {
        while (!isEnd) {
            CommonUtil.sleep(1000);
            FileUtil.writeStrToFile(new CpuInfo(device, pid).getProCpu() + "%" + "\n", resultFile);
        }
    }

    public void setIsEnd(boolean isEnd) {
        this.isEnd = isEnd;
    }
}

File: app/src/androidTest/kotlin/com/bbtest/perform/base/CpuThread.kt
package com.bbtest.perform.base

import androidx.test.uiautomator.UiDevice
import com.bbtest.utils.CommonUtil
import com.bbtest.utils.FileUtil
import java.io.File

class CpuThread(
    private val device: UiDevice,
    private val pid: Int,
    private val resultFile: File,
) : Thread() {
    private var isEnd = false

    override fun run() {
        while (!isEnd) {
            CommonUtil.sleep(1000)
            FileUtil.writeStrToFile(CpuInfo(device, pid).getProCpu().toString() + "%\n", resultFile)
        }
    }

    fun setIsEnd(isEnd: Boolean) {
        this.isEnd = isEnd
    }
}
